server/tests/classes.test.ts

```diff
@@ -0,0 +1,518 @@
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+
+            expect(response.status).toBe(400)
+            expect(response.body.error).toBe('No fields to update provided')
+        })
+
+        it('Invalid action, same fields without file provided', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            createClassMock.mockImplementation(async () => {
+                throw new Error('Invalid data')
+            })
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('description', 1)
+                .attach(
+                    'file',
+                    Buffer.from('test file content', 'utf-8'),
+                    'mockedFile1.txt'
+                )
+
+            expect(response.status).toBe(400)
+            expect(response.body.error).toBe('Invalid file type. Only quicktime,mp4 and mpeg video files are allowed.')
+        })
+
+        it('Should return unauthenticated error', async () => {
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .field('title', 'Clase 1')
+                .field('description', 'Descripción 1')
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
+                    Buffer.from('test file content', 'utf-8'),
+                    'mockedFile1.txt'
+                )
+
+            expect(response.status).toBe(401)
+        })
+
+        it('Should return unauthenticated error', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${UNAUTHORIZED_JWT2}`)
+                .field('title', 'Clase 1')
+                .field('order', 1)
+                .field('description', 'Descripción 1')
+                .attach('file', Buffer.from(''), {
+                    contentType: 'video/mp4',
+                    filename: 'mockedFile1.mp4',
+                });
+
+            expect(response.status).toBe(401)
+        })
+
+        it('Class not found', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve()
+            )
+
+            const response = await request(app)
+                .put('/api/v1/classes/615e2f3b1d9f9b2b4c9e9b1a')
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('description', 'Descripción 1')
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
```

Comment:
Delete this comment please. I understand that 'EVERYTHING' has already been done, although it is still better to put an mp4 file in the buffer, for code coherence